vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -164,6 +174,35 @@ void AdsImpl::InitializeStep4(
   ads_serve_->DownloadCatalog();
 }
 
+#if defined(OS_ANDROID)
+void AdsImpl::RemoveAllNotificationsAfterReboot() {
+  // ads notifications don't sustain reboot, so remove all
+  auto ads_shown_history = client_->GetAdsShownHistory();
+  if (!ads_shown_history.empty()) {
+    uint64_t ad_shown_timestamp =
+        ads_shown_history.front().timestamp_in_seconds;
+    uint64_t boot_timestamp = Time::NowInSeconds() -
+        static_cast<uint64_t>(base::SysInfo::Uptime().InSeconds());
+    if (ad_shown_timestamp <= boot_timestamp) {
```

Comment:
Can this logic to detect if all notifications should be removed be moved to a separate function as this will save the need for a comment (self-documented) and allow us to add a unit test for the check

Reply:
NONCRITICAL
Covered by: brave/brave-browser#6235 (17)